initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/documentation/RequestedDependenciesHelpDocumentCustomizer.java

```diff
@@ -44,20 +46,36 @@
 
 	private final String defaultPlatformVersion;
 
-	public RequestedDependenciesHelpDocumentCustomizer(ProjectDescription description, InitializrMetadata metadata) {
+	private final ProjectDescriptionDiff projectDescriptionDiff;
+
+	public RequestedDependenciesHelpDocumentCustomizer(
+			ProjectDescription description,
+			InitializrMetadata metadata,
+			ProjectDescriptionDiff projectDescriptionDiff) {
 		this.description = description;
 		this.metadata = metadata;
 		this.defaultPlatformVersion = metadata.getBootVersions().getDefault().getId();
+		this.projectDescriptionDiff = projectDescriptionDiff;
 	}
 
 	@Override
 	public void customize(HelpDocument document) {
+
 		this.description.getRequestedDependencies().forEach((id, dependency) -> {
 			Dependency dependencyMetadata = this.metadata.getDependencies().get(id);
 			if (dependencyMetadata != null) {
 				handleDependency(document, dependencyMetadata);
 			}
 		});
+
+		projectDescriptionDiff.ifVersionChanged(description, (oldVersion, newVersion) ->
```

Comment:
This will obviously not stay in the code, just adding in to show how it can be used. Also show the generic `ifPropertyChanged` method - we should not really need that but if someone did not want to provide a custom diff they could rely on that.